src/Microsoft.IdentityModel.Tokens/Validation/Validators.IssuerSigningKey.cs

```diff
@@ -117,9 +117,9 @@ internal static ValidationResult<ValidatedSigningKeyLifetime> ValidateIssuerSign
                             LogMessages.IDX10249,
                             LogHelper.MarkAsNonPII(notAfterUtc),
                             LogHelper.MarkAsNonPII(utcNow)),
-                        ValidationFailureType.SigningKeyValidationFailed,
                         typeof(SecurityTokenInvalidSigningKeyException),
-                        new StackFrame(true));
+                        new StackFrame(true),
+                        ValidationFailureType.SigningKeyValidationFailed);
```

Comment:
This could be a more specific ValidationFailureType.

Reply:
Looking into more specific errors in the extensibility PR.